exec/cnex/number.c

```diff
@@ -116,40 +118,39 @@ void number_toString(Number x, char *buf, size_t len)
             // Then copy the remainder of the mantissa to our [r]esult.
             memcpy(&r[count], mantissa, slen);
             // Ensure that [r]esult is null terminated.  I'm not 100% convinced this is necessary.
-            r[count+slen] = '\0';
+            r[slen += count] = '\0';
```

Comment:
There's no need to embed this expression inside an array access. Split into two statements.